### PR TITLE
medicationCC and dispenseRequest.numberOfRepeatsAllowed + fix reason code precedence

### PR DESCRIPTION
**Pull request by jgrant22**
from  [705](https://github.ibm.com/WH-Imaging/Clinical-Data-Pipeline/issues/705)(the last of 705)

reason Code seemed to be working correctly but the precedence was slightly off so I fixed that

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java [hunk 1]
```diff
@@ -624,9 +627,9 @@ void dispenseRequestTestRXE() {
         String hl7message = "MSH|^~\\&||||||S1|RDE^O11||T|2.6|||||||||\n"
                     + "PID|||1234^^^^MR||DOE^JANE^|||F||||||||||||||||||||||\n"
                     + "ORC|NW|||||E|10^BID^D4^^^R||20180622230000||||||||||||||||||||I\n"
-                    + "RXE|^Q24H&0600^^20210330144208^^ROU|DUONEB3INH^3 ML PLAS CONT : IPRATROPIUM-ALBUTEROL 0.5-2.5 (3) MG/3ML IN SOLN^ADS^^^^^^ipratropium-albuterol (DUONEB) nebulizer solution 3 mL|3||mL|47|||"
+                    + "RXE||DUONEB3INH^3 ML PLAS CONT : IPRATROPIUM-ALBUTEROL 0.5-2.5 (3) MG/3ML IN SOLN^ADS^^^^^^ipratropium-albuterol (DUONEB) nebulizer solution 3 mL|||||||"
                     // split and concatenate RXE for easier understanding
-                    + "|1|PC^^measureofunits||2213^ORDERING^PROVIDER||||||||||||||Wheezing^Wheezing^PRN||||^DUONEB|20180622230000||||||7|7|7\n"; // RXE 10 & 11 are on this line
+                    + "|1|PC^^measureofunits||0||||||||||||||||||||||||||7|\n"; // RXE 10 & 11 & 13 are on this line
```
Comment (cragun47) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java]: Thank you for simplifying data.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java [hunk 2]
```diff
@@ -616,6 +616,9 @@ void dispenseRequestTestRXO() {
         assertThat(disReq.getQuantity().getValue().toString()).isEqualTo("4.0");
         assertThat(disReq.getQuantity().getUnit()).isEqualTo("tablet");
         assertThat(disReq.getQuantity().getSystem()).isEqualTo("http://unitsofmeasure.org");
+
+        // dispenseRequest.NumberOfRepeatsAllowed RXO.13
+        assertThat(disReq.getNumberOfRepeatsAllowed()).hasToString("0");
```
Comment (cragun47) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java:621]: Thank you for reusing parses.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java [hunk 3]
```diff
@@ -281,7 +281,7 @@ void test_medicationCodeableConcept_authoredOn_and_intent_in_rde_with_just_rxe(S
         String hl7message = msh
                 + "PID|||1234^^^^MR||DOE^JANE^|||F||||||||||||||||||||||\n"
                 + "ORC|NW|F800006^OE|P800006^RX|||E|10^BID^D4^^^R||20190622160000\n"
-                + "RXE|^Q24H&0600^^20210330144208^^ROU|DUONEB3INH^3 ML PLAS CONT : IPRATROPIUM-ALBUTEROL 0.5-2.5 (3) MG/3ML IN SOLN^ADS^^^^^^ipratropium-albuterol (DUONEB) nebulizer solution 3 mL|3||mL|47||||1|PC||||||||||||||||||||^DUONEB|20180622230000||||||||\n";
+                + "RXE|||3||mL|47||||1|PC||||||||||||||||||||DUONEB3INH^3 ML PLAS CONT : IPRATROPIUM-ALBUTEROL 0.5-2.5 (3) MG/3ML IN SOLN^ADS^^^^^^ipratropium-albuterol (DUONEB) nebulizer solution 3 mL|20180622230000||||||||\n"; //RXE.31
```
Comment (cragun47) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java:284]: Which fields in RXE.31 are being checked?  There's a value in RXE.31.9 ??  Please remove the data or check it.

File: src/main/resources/hl7/resource/MedicationRequest.yml [hunk 4]
```diff
@@ -130,7 +138,7 @@ reasonCode:
   valueOf: datatype/CodeableConcept
   generateList: true
   expressionType: resource
-  specs: RXE.27 | RXO.20 | ORC.16
+  specs: RXO.20 | RXE.27 | ORC.16
```
Comment (cragun47) [inline src/main/resources/hl7/resource/MedicationRequest.yml:141]: Missing RXA-19 and test.
>MedicationRequest.reasonCode precedence: RXO-20, RXE-27, RXA-19, ORC-16.

**Review by cragun47 — CHANGES_REQUESTED** [review]
Two changes requested.

File: src/main/resources/hl7/resource/MedicationRequest.yml (hunk 4 above)
Comment (jgrant22) [reply src/main/resources/hl7/resource/MedicationRequest.yml:141]: None of the message types that we support create medication request from RXA segments

Comment (cragun47) [reply src/main/resources/hl7/resource/MedicationRequest.yml:141]: OK

**Review by cragun47 — APPROVED** [review]
Thanks for changes.

File: src/main/resources/hl7/resource/MedicationRequest.yml (hunk 4 above)
Comment (LisaWellman) [reply src/main/resources/hl7/resource/MedicationRequest.yml:141]: Concerned that we were able to change this order and no testcases broke.  There should have been a testcase verifying the precedence of reasonCode.

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java (hunk 3 above)
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java]: sorry I thought I checked this off

Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7MedicationRequestFHIRConversionTest.java]: this was taken care of

**Pull request merged**